chatter/src/components/Messages/components/Footer.js

```diff
@@ -1,17 +1,34 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
 
 const RETURN_KEY_CODE = 13;
 
-export default function Footer({ sendMessage, onChangeMessage, message }) {
+const Footer = React.memo(({ sendMessage }) => {
+  const inputElemRef = useRef();
+  const [canSend, setCanSend] = useState(false);
+
   const onKeyDown = ({ keyCode }) => {
-    if (keyCode !== RETURN_KEY_CODE ) { return; }
+    if (keyCode !== RETURN_KEY_CODE) { return; }
+
+    handleSendMessage();
+  }
 
-    sendMessage();
+  const handleSendMessage = () => {
+    const message = inputElemRef.current?.value;
+    if (message) {
+      sendMessage(message);
+      inputElemRef.current.value = '';
+      setCanSend(false);
+    }
+  }
+
+  const onChangeMessage = (e) => {
+    setCanSend(!!e.target.value);
```

Comment:
Refactored this `Footer` component so that when the chat thread updates, this component won't be re-rendered.